# Proyecto 2 - Proceso de generación de valor por extración de datos


Este proyecto esta diseñado para que usted pueda:

1.   Poner en práctica los conocimientos específicos del leguaje Python y las bibliotecas de Pandas, Matplotlib con Seaborn y numpy
2.   Generar una plantilla reutilizable de un proceso de limpieza, consolidación y carga de datos
3.   Aplicar los conocimientos analizando una base de datos del mundo real.

Utilizaremos el dataset superstore que puede encontar en la carpeta de datasets de Google Drive. Este es la información de ventas y ordenes de cuatro años de una tienda en línea y le permitirá hacer análisis de ventas, categorías etc.

Mantenga en mente que algunos errores en carga y limpieza de datos pueden surgir. Su tarea es identificarlos y sobreponerse a ellos.



In [1]:
from google.colab import drive

In [2]:
import pandas as pd

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


# Importar Datos e Identificar variables a limpiar

In [ ]:
df1 = pd.read_csv ( '/content/drive/MyDrive/ASOS/superstore/superstore_dataset2011-2015-1.csv',encoding= 'latin1')
df2 = pd.read_csv ( '/content/drive/MyDrive/ASOS/superstore/superstore_dataset2011-2015-2.csv',encoding= 'latin1')
df3 = pd.read_csv ( '/content/drive/MyDrive/ASOS/superstore/superstore_dataset2011-2015-3.csv',encoding= 'latin1')
df4 = pd.read_csv ( '/content/drive/MyDrive/ASOS/superstore/superstore_dataset2011-2015-4.csv',encoding= 'latin1')
df5 = pd.read_csv ( '/content/drive/MyDrive/ASOS/superstore/superstore_dataset2011-2015-5.csv',encoding= 'latin1')
df6 = pd.read_csv ( '/content/drive/MyDrive/ASOS/superstore/superstore_dataset2011-2015-8.csv',encoding= 'latin1')

In [ ]:
merged_df = pd.concat([df1, df2,df3,df4,df5,df6], ignore_index=True)

In [65]:
df.columns = df.columns.str.strip()

In [66]:
df.columns = df.columns.str.replace(' ', '_')

In [67]:
df.columns = df.columns.str.replace('-', '_')

In [68]:
df.columns = df.columns.str.lower()

In [69]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10244 entries, 0 to 10243
Data columns (total 25 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   unnamed:_0      10244 non-null  int64  
 1   row_id          10244 non-null  int64  
 2   order_id        10244 non-null  object 
 3   order_date      10244 non-null  object 
 4   ship_date       10244 non-null  object 
 5   ship_mode       10244 non-null  object 
 6   customer_id     10244 non-null  object 
 7   customer_name   10244 non-null  object 
 8   segment         10244 non-null  object 
 9   city            10244 non-null  object 
 10  state           10244 non-null  object 
 11  country         10244 non-null  object 
 12  postal_code     10244 non-null  int64  
 13  market          10244 non-null  object 
 14  region          10244 non-null  object 
 15  product_id      10244 non-null  object 
 16  category        10244 non-null  object 
 17  sub_category    10244 non-null 

In [ ]:
print(df.head())

# Limpieza de Datos

In [ ]:
duplicados = df[df.duplicated()]

In [ ]:
print(duplicados)

In [ ]:
df = df.drop_duplicates()

In [ ]:
df[df.duplicated(subset='postal_code', keep=False)]

In [ ]:
df['postal_code'] = df['postal_code'].fillna(0)

In [ ]:
df['postal_code'] = df['postal_code'].astype(int)

In [ ]:
df = df.drop_duplicates()

In [ ]:
df.category.head()

In [ ]:
df.category.unique()

In [ ]:
convertir_a_porcentaje = lambda x: "{:.2%}".format(x)
df['discount'] = df['discount'].apply(convertir_a_porcentaje)

In [ ]:
df.discount

In [ ]:
df['sales'] = df['sales'].astype(str).replace(',', '.', regex=True).astype(float)

In [ ]:
df.sales.head

<bound method NDFrame.head of 0        631.91364
1        437.72400
2        215.37600
3        264.16800
4        332.52960
           ...    
10255    198.64000
10256    217.18800
10257    381.78000
10258    179.46000
10259    138.83400
Name: sales, Length: 10260, dtype: float64>

In [ ]:
df['profit'] = df['profit'].astype(str).replace(',', '.', regex=True).astype(float)


In [ ]:
df['shipping_cost'] = df[ 'shipping_cost'].astype(str).replace(',', '.', regex=True).astype(float)

In [ ]:
df.profit

In [ ]:
df.shipping_cost

# Creación de nueva base de datos en Google Drive

In [ ]:
df.to_csv()

In [60]:
from google.colab import files

In [61]:
df.to_csv('tarea.csv')
files.download('tarea.csv')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

# Explorando datos

Conectar a la nueva base de datos

In [63]:
df = pd.read_csv ( '/content/drive/MyDrive/ASOS/superstore/superstore_date_2.csv')

In [64]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10244 entries, 0 to 10243
Data columns (total 25 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Unnamed: 0      10244 non-null  int64  
 1   row_id          10244 non-null  int64  
 2   order_id        10244 non-null  object 
 3   order_date      10244 non-null  object 
 4   ship_date       10244 non-null  object 
 5   ship_mode       10244 non-null  object 
 6   customer_id     10244 non-null  object 
 7   customer_name   10244 non-null  object 
 8   segment         10244 non-null  object 
 9   city            10244 non-null  object 
 10  state           10244 non-null  object 
 11  country         10244 non-null  object 
 12  postal_code     10244 non-null  int64  
 13  market          10244 non-null  object 
 14  region          10244 non-null  object 
 15  product_id      10244 non-null  object 
 16  category        10244 non-null  object 
 17  sub_category    10244 non-null 

In [70]:
df.drop('unnamed:_0', axis=1, inplace=True)

In [ ]:
df['order_date'] = pd.to_datetime(df['order_date'])

In [ ]:
df['ship_date'] = pd.to_datetime(df['ship_date'])

In [51]:
df['order_id'] = df['order_id'].astype('object')

In [ ]:
duplicados = df[df.duplicated(['order_id', 'product_id'], keep=False)]
print(duplicados)

In [53]:
df.drop_duplicates(['order_id', 'product_id'], keep=False, inplace=True)

# Analizando Categorías, tendencias, medias y creando valor

Su función será crear un "deep dive" que explorara 2 a 4 preguntas de negocios que pueda hacer con los datos: ¿Cuál es el mejor día de la semana de ventas? etc. Como mínimo su análisis debe incluir:

- Comparar dos variables con al menos una segmentación por categoría
- Analizar gráficamente multiples variables numéricas
- Plot charts, Gráficos de tendencia y Barra con su análisis de que significa
- Utilizar las ideas anteriores como sea necesario para cada pregunta de negocio.



In [74]:
import seaborn as sns

In [75]:
bestdate = df.groupby('order_date').size()


In [76]:
bestdate_sum = df.groupby('order_date').agg({'quantity': 'sum'})

In [77]:
fecha_mas_repetida = bestdate_sum ['quantity'].idxmax()

In [78]:
print( 'El dia con mas ventas: ', fecha_mas_repetida)

El dia con mas ventas:  2014-06-18 00:00:00


In [ ]:
print
(bestdate_sum)

quantity
order_date          
2011-01-14       127
2011-01-15        37
2011-01-16         5
2011-01-17        57
2011-01-18        17
...              ...
2014-12-14        38
2014-12-15       325
2014-12-16       219
2014-12-17       227
2014-12-18       222

[285 rows x 1 columns]

# Conclusiones

In [79]:
print( 'El dia con mas ventas: ', fecha_mas_repetida)

El dia con mas ventas:  2014-06-18 00:00:00


La fecha en la que los clientes realizaron mas compras fue  2014-06-18 pero el  dia que mas productos fueron vendidos fue el 2011-01-14 con un total de 127




In [80]:
print
(bestdate_sum)

quantity
order_date          
2011-01-14       127
2011-01-15        37
2011-01-16         5
2011-01-17        57
2011-01-18        17
...              ...
2014-12-14        38
2014-12-15       325
2014-12-16       219
2014-12-17       227
2014-12-18       222

[285 rows x 1 columns]